### PR TITLE
Add Electron menu

### LINKED ISSUE #10: Add Electron Menu
Currently we have removed custom menu by migrating `erb` to `cra-electron` in #5

### PR DESCRIPTION
**Pull request by veado**
Peek 2020-07-14 14-29

- [x] Create menu items
- [x] Support i18n for menu
- [x] Messaging `main` <-> `render` process via [ipc](https://www.electronjs.org/docs/api/ipc-main#ipcmain) (needed to support `lang` changes in menu)
- [x] Share `common` `i18n` sources
- [x] Build `electron` sources w/ `webpack` ~~+ `babel`~~

Closes #10

### CODE REVIEW

File: electron/electron.ts [hunk 1]
```diff
@@ -53,15 +60,36 @@ const closeHandler = () => {
...
+  const { default: install, REACT_DEVELOPER_TOOLS } = require('electron-devtools-installer')
   try {
     await install(REACT_DEVELOPER_TOOLS)
-    await install(REDUX_DEVTOOLS)
   } catch (e) {
     warn('unable to install devtools', e)
   }
 }
 
+enum IPCMessages {
```
Comment (veado) [inline electron/electron.ts:72]: Move it into `src/share

File: src/services/i18n/service.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+const LOCALE_KEY = 'asgdx-locale'
+
+const defaultLocale = () => {
+  const lang = navigator.language.split(/[-_]/)[0]
+  getLocaleFromString(lang)
+}
+export const initialLocale = (): Locale => {
+  const locale = (localStorage?.getItem(LOCALE_KEY) as Locale) || defaultLocale()
+  ipcRenderer.send('UPDATE_LANG', locale)
```
Comment (veado) [inline src/services/i18n/service.ts:15]: `IPCMessages.UPDATE_LANG`

File: src/services/i18n/service.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+/**
+ * Observable state of `Locale`
+ */
+const { get$: locale$, get: locale, set: setLocale } = observableState<Locale>(initialLocale())
+
+const changeLocale = (next: Locale) => {
+  if (next !== locale()) {
+    localStorage.setItem(LOCALE_KEY, next)
+    ipcRenderer.send('UPDATE_LANG', next)
```
Comment (veado) [inline src/services/i18n/service.ts:27]: `IPCMessages.UPDATE_LANG`

File: electron/electron.ts [hunk 4]
```diff
@@ -1,12 +1,19 @@
...
 
-import { BrowserWindow, app, remote } from 'electron'
+import { BrowserWindow, app, remote, ipcMain } from 'electron'
 import electronDebug from 'electron-debug'
 import isDev from 'electron-is-dev'
 import log from 'electron-log'
 import { warn } from 'electron-log'
+import { fromEvent } from 'rxjs'
 
+// import { IS_PRODUCTION } from '../src/shared/const'
```
Comment (thatStrangeGuyThorchain) [inline electron/electron.ts:10]: comments

File: electron/electron.ts [hunk 5]
```diff
@@ -1,12 +1,19 @@
...
+import IPCMessages from '../src/shared/ipc/messages'
+import { setMenu } from './menu'
+
+// export const IS_DEV = isDev && IS_PRODUCTION
```
Comment (thatStrangeGuyThorchain) [inline electron/electron.ts:14]: same here

**Pull request merged**